### PR TITLE
Slash commands (Fix #13)

### LINKED ISSUE #13: Use new Discord slash commands

### PR DESCRIPTION
**Pull request by AFCMS**
- [x] Slash commands instead of normal ones
    - [x] `/mute`
    - [x] `/unmute`
    - [x] `/x`
    - [x] `/rank` (with autocomplete)
    - [x] `/leaders`
- [x] Update to discord.js 14
- [x] Switch from CJS to ESM
- [x] Commands privilege check at the Discord level
- [x] More Embeds
- [x] Fix broken leaderboard 
- [x] README

Fix https://github.com/MT-CTF/ctf-discord-bot/issues/13

**MERGE AS A SINGLE COMMIT WHEN READY**

### CODE REVIEW

**Comment by AFCMS** [comment]
For some reason the leaderboard code is broken (I didn't touched it of course)
All other stuff works

**Comment by AFCMS** [comment]
The /rank command no longer allows mentions and player names, but works only with player names, which are auto-completed.
When using with no params it works the same as before.

**Comment by AFCMS** [comment]
@LoneWolfHT

File: index.js [hunk 1]
```diff
@@ -1,270 +1,474 @@
...
-			staffMessages = [];
-		}
+	await discordClient.login(token)
+	await redisClient.connect()
+
+	await updateRankings()
+	setInterval(updateRankings, 6000)
+
+	http
+		.createServer(function (req, res) {
```
Comment (LoneWolfHT) [inline index.js]: This indentation looks wrong

Comment (AFCMS) [reply index.js]: Prettier's fault

Will fix

Comment (AFCMS) [reply index.js]: Fixed

**Comment by AFCMS** [comment]
@LoneWolfHT Could you consider re-reviewing the PR?

**Review by LoneWolfHT — APPROVED** [review]

**Pull request merged**